rotkehlchen/exchanges/bit2me.py

```diff
@@ -0,0 +1,1014 @@
+            side = raw_trade['side']
+
+            # Determine spend/receive based on side
+            is_buy = side.lower() == 'buy'
+            spend, receive = get_swap_spend_receive(
+                is_buy=is_buy,
+                base_asset=base_asset,
+                quote_asset=quote_asset,
+                amount=amount,
+                rate=price,
+            )
+
+            # Parse fee
+            fee_amount = deserialize_fval_or_zero(raw_trade.get('fee', '0'))
+            fee_asset_symbol = raw_trade.get('feeAsset', quote_symbol)
+            fee_asset = asset_from_bit2me(fee_asset_symbol)
+
+            # Create swap events
+            return create_swap_events(
+                spend=spend,
+                receive=receive,
+                timestamp=timestamp_ms,
+                location=self.location,
+                location_label=self.name,
+                fee=AssetAmount(asset=fee_asset, amount=fee_amount),
+                group_identifier=create_group_identifier_from_unique_id(self.location, trade_id),
+            )
+
+        except KeyError as e:
+            raise DeserializationError(
+                f'Missing key in Bit2me trade: {e}. Raw trade: {raw_trade}',
+            ) from e
+
+    def _deserialize_brokerage_trade(
+        self,
+        raw_transaction: dict[str, Any],
+    ) -> list[SwapEvent]:
+        """Deserialize a brokerage purchase/sale from Bit2me transaction API.
+
+        Brokerage transactions have type="transfer" and subtype="purchase".
+        They represent direct crypto purchases with fiat or crypto-to-crypto swaps.
+
+        Expected structure:
+        {
+            "id": "transaction_id",
+            "date": "2025-11-18T23:33:51.381Z",
+            "type": "transfer",
+            "subtype": "purchase",
+            "origin": {
+                "amount": "100.00000000",
+                "currency": "EUR"
+            },
+            "destination": {
+                "amount": "0.00122130",
+                "currency": "BTC"
+            }
+        }
+
+        Args:
+            raw_transaction: Raw transaction data from API
+
+        Returns:
+            List of SwapEvent(s) representing the trade
+
+        Raises:
+            DeserializationError: If deserialization fails
+            UnknownAsset: If asset cannot be resolved
+            UnsupportedAsset: If asset is not supported
+        """
+        try:
+            # Extract transaction ID
+            trade_id = str(raw_transaction['id'])
+
+            # Parse timestamp from ISO format
+            timestamp_str = raw_transaction['date']
+            from datetime import datetime
+
+            timestamp_str_clean = (
+                timestamp_str[:-1] + '+00:00' if timestamp_str.endswith('Z') else timestamp_str
+            )
+            dt = datetime.fromisoformat(timestamp_str_clean)
+            timestamp_ms = TimestampMS(int(dt.timestamp() * 1000))
```

Comment:
check `deserialize_timestamp_from_date`. I believe it covers this logic that you have here